### ISSUE TITLE
warning: --filter is ignored in local clones; use file:// instead.

### ISSUE THREAD

**Issue by HaleTom** [Morganamilo/paru#1104]
**Have you checked the readme and man page for this feature?**
Yes

**Have you checked previous issues for this feature?**
Yes

## TL;DR:

Please use `file://` as the git clone source source when doing:
```
==> Extracting sources...
  -> Creating working copy of groot git repo...
```

## Detail

I'm getting the following warning when installing git packages (eg: `groot`):
```
export GITFLAGS=--filter=blob:none
paru -S groot
```

This produces the output:

```
==> Extracting sources...
  -> Creating working copy of groot git repo...
Cloning into 'groot'...
warning: --filter is ignored in local clones; use file:// instead.
done.
remote: Enumerating objects: 1, done.
remote: Counting objects: 100% (1/1), done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (1/1), 918 bytes | 918.00 KiB/s, done.
remote: Enumerating objects: 1, done.
remote: Counting objects: 100% (1/1), done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (1/1), 565 bytes | 565.00 KiB/s, done.
remote: Enumerating objects: 1, done.
remote: Counting objects: 100% (1/1), done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (1/1), 355 bytes | 355.00 KiB/s, done.
remote: Enumerating objects: 1, done.
remote: Counting objects: 100% (1/1), done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (1/1), 769 bytes | 769.00 KiB/s, done.
remote: Enumerating objects: 1, done.
remote: Counting objects: 100% (1/1), done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (1/1), 986 bytes | 986.00 KiB/s, done.
remote: Enumerating objects: 1, done.
remote: Counting objects: 100% (1/1), done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (1/1), 1.06 KiB | 1.06 MiB/s, done.
remote: Enumerating objects: 1, done.
<snip>
```

The git output continues S L O W L Y (at about one "Receiving" per second due to network access).  This is not part of the clone operation, but it happens when operating on the imporperly cloned repository after the clone finished note the clone's `done` output).  Git has /many/ promised objects it doesn't have locally, and tries to receive them from the remote.

## Proof of fix

As suggested, using `file:///path/to/source/directory` works with `--filter=blob:none`.

First, a failing example without `file://`:

```
% GITFLAGS=--filter=blob:none  git clone ~/bin testx
Cloning into 'testx'...
warning: --filter is ignored in local clones; use file:// instead.
done.
```

Now without the warning, using `file://`:

```
% GITFLAGS=--filter=blob:none  git clone file://$HOME/bin testxy
Cloning into 'testxy'...
remote: Enumerating objects: 756, done.
remote: Counting objects: 100% (756/756), done.
remote: Compressing objects: 100% (423/423), done.
remote: Total 756 (delta 378), reused 696 (delta 329), pack-reused 0
Receiving objects: 100% (756/756), 100.15 KiB | 14.31 MiB/s, done.
Resolving deltas: 100% (378/378), done.
remote: Enumerating objects: 166, done.
remote: Counting objects: 100% (166/166), done.
remote: Compressing objects: 100% (154/154), done.
remote: Total 166 (delta 0), reused 148 (delta 0), pack-reused 0
Receiving objects: 100% (166/166), 98.77 KiB | 16.46 MiB/s, done.
```

## Related
Support git blobless clones #1027
